spec/requests/posts_spec.rb

```diff
@@ -2,16 +2,28 @@
 
 RSpec.describe 'Posts', type: :request do
   describe 'GET /index' do
-    it 'returns http success' do
-      get '/posts/index'
-      expect(response).to have_http_status(:success)
+    before(:example) { get '/users/1/posts' }
+    it 'responds with the code 200' do
+      expect(response).to have_http_status(:ok)
+    end
+    it "renders 'index' template" do
+      expect(response).to render_template(:index)
+    end
```

Comment:
- [ ] All of your tests which render a view are not working. I noticed that the problem was the absence of the `rails-controller-testing` gem. Kindly add this gem to your `Gemfile`, then run `bundle install` and everything should be fine 👍

Reply:
Thanks a lot @codepantha